File: learning-spark-exercise/src/main/scala/spark/scala/BasicLoadSequenceFile.scala
package spark.scala

import org.apache.spark._
import org.apache.hadoop.io.{ IntWritable, Text }

/**
 * Example 5-21. Loading a SequenceFile in Scala
 */
object BasicLoadSequenceFile {
  def main(args: Array[String]): Unit = {
    val master = args(0)
    val inFile = args(1)
    val sc = new SparkContext(master, "", System.getenv("SPARK_HOME"))
    val data = sc.sequenceFile(inFile, classOf[Text], classOf[IntWritable]).map { case (x, y) => (x.toString, y.get()) }

  }
}

File: learning-spark-exercise/src/main/java/spark/java/BasicLoadSequenceFile.java
package spark.java;

import org.apache.hadoop.io.IntWritable;
import org.apache.hadoop.io.Text;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.PairFunction;

import scala.Tuple2;



/**
 * Example 5-22. Loading a SequenceFile in Java
 * 
 * @author pzhang1
 *
 */
public class BasicLoadSequenceFile {
	
	public static class ConvertToNativetypes implements PairFunction<Tuple2<Text,IntWritable>,String,Integer>{

		@Override
		public Tuple2<String, Integer> call(Tuple2<Text, IntWritable> record)
				throws Exception {
			return new Tuple2(record._1.toString(),record._2.get());
		}
		
	}
	
	public static void main(String[] args) throws Exception {
		if (args.length != 2) {
			throw new Exception(
					"Usage BasicLoadSequenceFile [sparkMaster] [input]");
		}
		String master = args[0];
		String fileName = args[1];

		JavaSparkContext sc = new JavaSparkContext(master,
				"basicloadsequencefile", System.getenv("SPARK_HOME"),
				System.getenv("JARS"));
		
		JavaPairRDD<Text,IntWritable> input = sc.sequenceFile(fileName, Text.class, IntWritable.class);
		JavaPairRDD<String,Integer> result = input.mapToPair(new ConvertToNativetypes());
	}

}
